### PR TITLE
parameterize TableRead with a TableReader

### PR DESCRIPTION
**Pull request by cseed**
convert TableImport into TextTableReader

### CODE REVIEW

**Comment by jigold** [comment]
This looks good to me. There's a bunch of test failures mainly in TextTableSuite.

**Comment by cseed** [comment]
Finally tracked down on the failures.  This should be ready to go now.

File: hail/python/hail/ir/table_reader.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                 skip_blank_lines, force_bgz):
+        self.config = {
+            'files': paths,
+            'typeMapStr': {f: t._parsable_string() for f, t in types.items()},
+            'comment': comment,
+            'separator': delimiter,
+            'missing': missing,
+            'noHeader': no_header,
+            'impute': impute,
+            'nPartitions': None,
```
Comment (jigold) [inline hail/python/hail/ir/table_reader.py]: min_partitions isn't used in the config at all? Is it supposed to go here?

Comment (cseed) [reply hail/python/hail/ir/table_reader.py]: Oops, good catch, it was supposed to go here.

**Pull request merged**